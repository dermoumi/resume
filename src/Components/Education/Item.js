import React, { Component, Fragment } from 'react'
import styled from 'styled-components'

const StyleBase = styled.article`
  margin-bottom: 16px;

  &:last-child {
    margin-bottom: 0;
  }

  .title {
    font-size: 1.1em;
    letter-spacing: 1px;
    line-height: 1.5rem;
    color: ${props => props.theme.primary};
  }

  .sub-title {
    font-size: 0.9em;
    line-height: 1.25rem;
    text-transform: uppercase;
    color: ${props => props.theme.secondary};
  }

  .period {
    font-weight: 400;
    margin-left: 8px;
    white-space: nowrap;
    color: ${props => props.theme.accent};
  }

  .period-arrow {
    font-family: icomoon;
    font-size: 0.9em;
  }
`

export default class Item extends Component {
  render() {
    let period
    if (
      Array.isArray(this.props.period) &&
      typeof this.props.period[0] === 'string'
    ) {
      period = (
        <Fragment>
          {this.props.period[0]}
          <span className="period-arrow"> &#xe900; </span>
          {this.props.period[1]}
        </Fragment>
      )
    } else {
      period = this.props.period
    }

    return (
      <StyleBase {...this.props}>
        <div className="heading">
          <div className="title">{this.props.degree}</div>
          <div className="sub-title">
            {this.props.info} <span className="period">{period}</span>
          </div>
        </div>
      </StyleBase>
    )
  }
}
