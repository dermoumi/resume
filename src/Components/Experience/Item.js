import React, { Component, Fragment } from 'react'
import styled from 'styled-components'

const StyleBase = styled.article`
  margin-bottom: 16px;
  /* max-width: 608px; */

  &:last-child {
    margin-bottom: 0;
  }

  .label {
    font-size: 1.1em;
    letter-spacing: 1px;
    line-height: 1.25rem;
    color: ${props => props.theme.primary};
  }

  .sub-title {
    font-size: 0.9em;
    line-height: 1.25rem;
    text-transform: uppercase;
    color: ${props => props.theme.secondary};
  }

  .label a,
  .sub-title a {
    color: inherit;
    text-decoration: none;

    &:hover {
      text-decoration: underline;
    }

    &:after {
      font-family: icomoon;
      font-size: 0.9em;
      content: '\\e907';
      margin-left: 5px;

      @media print {
        display: none;
      }
    }
  }

  .period {
    font-weight: 400;
    white-space: nowrap;
    color: ${props => props.theme.accent};
  }

  .period-arrow {
    font-family: icomoon;
    font-size: 0.9em;
  }

  .item-content {
    line-height: 1.25rem;
    padding-top: 4px;

    ul {
      list-style: none;
      margin: 0;
      padding: 0;
      padding-left: calc(1em + 10px);

      li {
        margin-bottom: 6px;

        &:before {
          font-family: icomoon;
          content: '\\e906';
          color: ${props => props.theme.accent};
          margin-left: calc(-1em - 10px);
          margin-right: 10px;
        }
      }
    }
  }
`

export default class Item extends Component {
  render() {
    let period
    if (Array.isArray(this.props.period)) {
      period = (
        <Fragment>
          {this.props.period[0]}
          <span className="period-arrow"> &#xe900; </span>
          {this.props.period[1]}
        </Fragment>
      )
    } else {
      period = this.props.period
    }

    const tasks =
      this.props.tasks &&
      this.props.tasks.map((task, i) => (
        <li key={i} dangerouslySetInnerHTML={{ __html: task }} />
      ))

    return (
      <StyleBase {...this.props}>
        <div className="heading">
          <div className="label">
            {this.props.labelLink ? (
              <a
                href={this.props.labelLink}
                target="_blank"
                rel="noreferrer noopener"
              >
                {this.props.label}
              </a>
            ) : (
              this.props.label
            )}
          </div>
          <div className="sub-title">
            {this.props.subtitleLink ? (
              <a
                href={this.props.subtitleLink}
                target="_blank"
                rel="noreferrer noopener"
              >
                {this.props.subtitle}
              </a>
            ) : (
              this.props.subtitle
            )}{' '}
            <span className="period">{period}</span>
          </div>
        </div>
        {tasks && (
          <div className="item-content">
            <ul>{tasks}</ul>
          </div>
        )}
      </StyleBase>
    )
  }
}
