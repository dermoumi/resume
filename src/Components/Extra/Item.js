import React, { Component } from 'react'
import styled from 'styled-components'

const StyleBase = styled.article`
  margin-bottom: 16px;

  &:last-child {
    margin-bottom: 0;
  }

  .label {
    font-size: 1.1em;
    letter-spacing: 1px;
    line-height: 1.25rem;
    color: ${props => props.theme.primary};
  }

  .sub-title {
    font-size: 0.9em;
    line-height: 1.25rem;
    text-transform: uppercase;
    color: ${props => props.theme.secondary};
  }
`

export default class Item extends Component {
  render() {
    return (
      <StyleBase {...this.props}>
        <div className="heading">
          <div className="label">{this.props.label}</div>
          <div className="sub-title">{this.props.details}</div>
        </div>
      </StyleBase>
    )
  }
}
