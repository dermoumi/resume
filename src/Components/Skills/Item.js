import React, { Component } from 'react';
import styled from 'styled-components';

const StyleBase = styled.article`
  margin-bottom: 16px;

  &:last-child {
    margin-bottom: 0;
  }

  .title {
    font-size: 1.1em;
    letter-spacing: 1px;
    line-height: 1.5rem;
    color: ${props => props.theme.primary};
  }

  .sub-title {
    font-size: 0.9em;
    line-height: 1.25rem;
    text-transform: uppercase;
    color: ${props => props.theme.secondary};
  }

  .level {
    font-weight: 400;
    margin-left: 4px;
    font-size: 0.8em;
    letter-spacing: 0;
    white-space: nowrap;
    color: ${props => props.theme.accent};
  }
`;

export default class Item extends Component {
  render() {
    return (<StyleBase {...this.props}>
      <div className="heading">
        <div className="title">{this.props.label} <span className="level">{this.props.level}</span></div>
        <div className="sub-title">{this.props.details}</div>
      </div>
    </StyleBase>);
  }
}
