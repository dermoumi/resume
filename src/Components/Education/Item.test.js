import React from 'react'
import { shallow, mount } from 'enzyme'
import Item from './Item'

it('renders education item degree', () => {
  const degree = 'My degree'
  const wrapper = shallow(<Item degree={degree} />)

  expect(wrapper).toMatchSnapshot()
  expect(wrapper.text()).toContain(degree)
})

it('renders education item info', () => {
  const info = 'My info'
  const wrapper = shallow(<Item info={info} />)

  expect(wrapper).toMatchSnapshot()
  expect(wrapper.text()).toContain(info)
})

it('renders education item year', () => {
  const year = 1337
  const wrapper = shallow(<Item period={year} />)

  const periodSpan = wrapper.find('span.period')
  expect(wrapper).toMatchSnapshot()
  expect(periodSpan.exists()).toBe(true)
  expect(periodSpan.text()).toContain(year)
})

it('renders education item period', () => {
  const period = [1337, '20XX']
  const wrapper = shallow(<Item period={period} />)

  const periodSpan = wrapper.find('span.period')
  expect(wrapper).toMatchSnapshot()
  expect(periodSpan.exists()).toBe(true)
  expect(periodSpan.text()).toMatch(
    new RegExp(`${period[0]}\\s+.\\s+${period[1]}`)
  )
})

it('renders all info correctly', () => {
  const props = {
    degree: 'My degree',
    info: 'My info',
    period: [1337, '20XX'],
  }

  const wrapper = mount(<Item {...props} />)
  expect(wrapper).toMatchSnapshot()
})
