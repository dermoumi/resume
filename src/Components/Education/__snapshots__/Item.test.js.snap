// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Education section item respects the theme 1`] = `
.c0 {
  margin-bottom: 16px;
}

.c0:last-child {
  margin-bottom: 0;
}

.c0 .title {
  font-size: 1.1em;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  line-height: 1.25rem;
  color: #FFCC00;
}

.c0 .sub-title {
  font-size: 0.9em;
  line-height: 1.25rem;
  text-transform: uppercase;
  color: #00FFCC;
}

.c0 .period {
  font-weight: 400;
  white-space: nowrap;
  color: #CC00FF;
}

.c0 .period-arrow {
  font-family: icomoon;
  font-size: 0.9em;
}

<Item
  degree="My degree"
  info="My info"
  period={
    Array [
      1337,
      "20XX",
    ]
  }
  theme={
    Object {
      "accent": "#CC00FF",
      "primary": "#FFCC00",
      "secondary": "#00FFCC",
    }
  }
>
  <styled.article
    degree="My degree"
    info="My info"
    period={
      Array [
        1337,
        "20XX",
      ]
    }
    theme={
      Object {
        "accent": "#CC00FF",
        "primary": "#FFCC00",
        "secondary": "#00FFCC",
      }
    }
  >
    <StyledComponent
      degree="My degree"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  margin-bottom: 16px;

  &:last-child {
    margin-bottom: 0;
  }

  .title {
    font-size: 1.1em;
    letter-spacing: 1px;
    line-height: 1.25rem;
    color: ",
              [Function],
              ";
  }

  .sub-title {
    font-size: 0.9em;
    line-height: 1.25rem;
    text-transform: uppercase;
    color: ",
              [Function],
              ";
  }

  .period {
    font-weight: 400;
    white-space: nowrap;
    color: ",
              [Function],
              ";
  }

  .period-arrow {
    font-family: icomoon;
    font-size: 0.9em;
  }
",
            ],
          },
          "displayName": "styled.article",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "article",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      info="My info"
      period={
        Array [
          1337,
          "20XX",
        ]
      }
      theme={
        Object {
          "accent": "#CC00FF",
          "primary": "#FFCC00",
          "secondary": "#00FFCC",
        }
      }
    >
      <article
        className="c0"
      >
        <div
          className="heading"
        >
          <div
            className="title"
          >
            My degree
          </div>
          <div
            className="sub-title"
          >
            My info
             
            <span
              className="period"
            >
              1337
              <span
                className="period-arrow"
              >
                 î¤€ 
              </span>
              20XX
            </span>
          </div>
        </div>
      </article>
    </StyledComponent>
  </styled.article>
</Item>
`;
